app/uk/gov/hmrc/crdlcache/config/ListConfig.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.crdlcache.config
+
+import uk.gov.hmrc.crdlcache.models.{CodeListCode, CodeListOrigin}
+
+trait ListConfig {
+  def code: CodeListCode
+  def origin: CodeListOrigin
+  def keyProperty: String
+}
```

Comment:
One thing that I particularly don't enjoy here is the duplication of the concept of standard vs correspondence codelists in both `ListConfig` and `CodeListType`.

I experimented with removing `ListConfig` and adding a `def valueProperty: Option[String]` to `CodeListConfig` then using the `CodeListType` to change the behaviour when converting the DPS data, but I didn't love the result.

Would love any ideas that you have about improving this.